[TechDocs] Accept sequence node custom tags in mkdocs config validation

viktorbahr opened the pull request:
## Hey, I just made a Pull Request!

This PR fixes #10695, making it possible to not only use custom tags with scalar node type, but also with sequence node type (such as the `!ENV` tag) in the mkdocs config.

#### :heavy_check_mark: Checklist

<!--- Please include the following in your Pull Request when applicable: -->

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Tests for new functionality and regression tests for bug fixes
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

iamEAP reviewed (commented):
Thanks for the detailed issue and PR, @viktorbahr!  Use-case makes a lot of sense.  I'd not seen this capability in `mkdocs` before, and it gives me ideas for some improvements we can do in our internal systems too 😬 

I'm good with these changes conceptually!  Just a couple of minor things to take care of to get it in.  Comments below.

viktorbahr commented on plugins/techdocs-node/src/stages/generate/__fixtures__/mkdocs_with_env_tag.yml:
```diff
@@ -0,0 +1 @@
+site_name: !ENV [SITE_NAME, OTHER_NAME, 'My default site name']
```
Will do :ok_hand:

viktorbahr commented on .changeset/nice-boats-wonder.md:
```diff
@@ -0,0 +1,5 @@
+---
```
Sure, will change that.

viktorbahr commented:
> Thanks for the detailed issue and PR, @viktorbahr! Use-case makes a lot of sense. I'd not seen this capability in `mkdocs` before, and it gives me ideas for some improvements we can do in our internal systems too grimacing
> 
> I'm good with these changes conceptually! Just a couple of minor things to take care of to get it in. Comments below.

Glad to hear that you find my contribution useful @iamEAP and thanks for your feedback - I've addressed the points mentioned by you.

iamEAP reviewed (approved):
💯 Thanks again @viktorbahr.
